zp-core/template-functions.php

```diff
@@ -4063,6 +4063,9 @@ function checkForGuest(&$hint = NULL, &$show = NULL) {
 function checkAccess(&$hint = NULL, &$show = NULL) {
 	global $_zp_current_album, $_zp_current_search, $_zp_gallery, $_zp_gallery_page,
 	$_zp_current_zenpage_page, $_zp_current_zenpage_news;
+	if (isset($_GET['download']) && extensionEnabled('downloadList')) {
+		return false; // Handled by downloadList extension
```

Comment:
If the plugin is enabled and a download is requested, zp-core/index.php will serve the standard password page unless the plugin, after checking the download credentials, provides a download or an error page